### PR TITLE
Albert Stevens

### PR DESCRIPTION
**Pull request by akorolczuk**
https://bitbucket.org/mertstevens/albert_m_stevens_tank_battalion/src/master/

### CODE REVIEW

File: AlbertStevens/AIController.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(TankBase))]
+public class AIController : MonoBehaviour
+{
+    public float changeDirectionDecisionChance = 0.02f;
+    private TankBase tankBase;
+    private FacingDirection fd;
+
```
Comment (akorolczuk) [inline AlbertStevens/AIController.cs]: Not very efficient way of doing this

**Comment by akorolczuk** [comment]
"No" from me.

File: AlbertStevens/GridManager.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+            bool stillAlive = true;
+
+            foreach(var actor in actors)
+            {
+                if (actor is IDamageable && actor.faction != projectile.faction)
+                {
+                    foreach (var cell in cells)
+                    {
+                        if (actor.GetOccupyingCells().Contains(cell))
```
Comment (Moniraego) [inline AlbertStevens/GridManager.cs]: For all projectiles, for every cell they occupy, go through all actors and see if they occupy that cell. Pretty inefficient, but might be using the same pragmatic logic as the top of this function states.

**Comment by Moniraego** [comment]
This is way more succinct than Adam but also more hacky. The code seems pretty disorganised. Then again, he mentions several times in comments 'I would do it like this if I had time'. Given the circumstances it's hard to criticise.

File: AlbertStevens/GridManager.cs (hunk 2 above)
Comment (akorolczuk) [reply AlbertStevens/GridManager.cs]: also method long af

**Comment by akorolczuk** [comment]
Ok we played the game and it's actually pretty cool. Let's have him over.

**Pull request closed without merging**